lib/schema.js

```diff
@@ -500,33 +500,118 @@ const addCategoryData = (ms, schemas) => {
   return errors;
 };
 
+const extractActions = schemas => {
+  for (const { name, definition } of schemas.categories) {
+    Object.entries(definition).forEach(([key, value]) => {
+      const decorator = extractDecorator(value);
+      if (decorator === 'Action') {
+        delete definition[key];
+        schemas.actions.push({
+          name: key,
+          category: name,
+          definition: value,
+        });
+      }
+    });
+  }
+};
+
+const typeToProps = {
+  category: 'categories',
+  domains: 'domains',
```

Comment:
Is it intentional that the name on the left hand side is in plural too?

Reply:
Yes, because there may be many domains in every object.